todo_app/app.py

```diff
@@ -1,46 +1,104 @@
+import os
 from flask import Flask, render_template, redirect, url_for, request
+from functools import wraps
+import requests
+from flask_login import LoginManager, login_required, UserMixin, login_user, current_user
+from oauthlib.oauth2 import WebApplicationClient
 
 from todo_app.data.mongodb import MongoDB
+from todo_app.user import User
 from todo_app.view_model import ViewModel
 
+def writer_required(f):
+    @login_required
+    @wraps(f)
+    def decorated_function(*args, **kwargs):
+        user = User(current_user.id)
+        if not user.is_writer():
+            flash("You do not have permisions to perform this action")
```

Comment:
Here, I don't think you've tested this route; `flash` is not imported, and the `return None` means it wouldn't  have an effect anyway (and in fact just errors). One solution might be to import `abort` from flask, and `return abort(401, 'Permission Denied')`